presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestJdbcPreparedStatement.java

```diff
@@ -0,0 +1,190 @@
+    public void teardown()
+    {
+        closeQuietly(server);
+    }
+
+    @Test
+    public void testExecuteQuery()
+            throws Exception
+    {
+        try (Connection connection = createConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        "SELECT ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?")) {
+            setTestValues(statement);
+
+            try (ResultSet rs = statement.executeQuery()) {
+                assertTrue(rs.next());
+                assertTestValues(rs);
+                assertFalse(rs.next());
+            }
+        }
+    }
+
+    @Test
+    public void testExecuteUpdate()
+            throws Exception
+    {
+        try (Connection connection = createConnection("blackhole", "blackhole")) {
+            try (Statement statement = connection.createStatement()) {
+                statement.execute("CREATE TABLE test_execute_update (" +
+                        "c_boolean boolean, " +
+                        "c_tinyint tinyint, " +
+                        "c_smallint smallint, " +
+                        "c_integer integer, " +
+                        "c_bigint bigint, " +
+                        "c_real real, " +
+                        "c_double double, " +
+                        "c_decimal decimal, " +
+                        "c_varchar varchar, " +
+                        "c_varbinary varbinary, " +
+                        "c_date date, " +
+                        "c_time time, " +
+                        "c_timestamp timestamp," +
+                        "c_null bigint)");
+            }
+
+            try (PreparedStatement statement = connection.prepareStatement(
+                    "INSERT INTO test_execute_update VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)")) {
+                setTestValues(statement);
+                assertEquals(statement.executeUpdate(), 1);
+            }
+        }
+    }
+
+    @Test
+    public void testPrepareMultiple()
+            throws Exception
+    {
+        try (Connection connection = createConnection();
+                PreparedStatement statement1 = connection.prepareStatement("SELECT 123");
+                PreparedStatement statement2 = connection.prepareStatement("SELECT 456")) {
+            try (ResultSet rs = statement1.executeQuery()) {
+                assertTrue(rs.next());
+                assertEquals(rs.getLong(1), 123);
+                assertFalse(rs.next());
+            }
+
+            try (ResultSet rs = statement2.executeQuery()) {
+                assertTrue(rs.next());
+                assertEquals(rs.getLong(1), 456);
+                assertFalse(rs.next());
+            }
+        }
+    }
+
+    private static void setTestValues(PreparedStatement statement)
+            throws SQLException
+    {
+        statement.setBoolean(1, true);
+        statement.setByte(2, (byte) 2);
+        statement.setShort(3, (short) 3);
+        statement.setInt(4, 4);
```

Comment:
`com.facebook.presto.jdbc.PrestoPreparedStatement#setObject(int, java.lang.Object)` and `com.facebook.presto.jdbc.PrestoPreparedStatement#setObject(int, java.lang.Object, int)` (the conversions) are not tested. You could have a variant of `setTestValues` that sets all values using those alternative methods. Or, better yet, split the cases as in `com.facebook.presto.jdbc.TestJdbcResultSet#testObjectTypes`. In pseudo-code:
```
testParameter( expectedValue, setter_or_binder ) {
  ps = prepare("select ?");
  if (setter_or_binder instanceof TriConsumer) {
    setter_or_binder.apply(ps, 1, expectedValue)
  } else { 
    setter_or_binder.apply(ps, 1)
  }

  rs = ps.execute
  rs.next
  assertEq(expectedValue, rs.getObject());
}
```
and then usage:
```
testParameter(1L, PreparedStatement::setLong);
testParameter(1L, (ps, index) -> ps.setObject(1L));
testParameter(1L, (ps, index) -> ps.setObject(index, 1, Types.BIGINT));
testParameter(1L, (ps, index) -> ps.setObject(index, 1L, Types.BIGINT));
testParameter(1L, (ps, index) -> ps.setObject(index, "1", Types.BIGINT));
```